packages/driver/cypress/integration/e2e/multi-domain/multi_domain_config_env_spec.ts

```diff
@@ -0,0 +1,286 @@
+        // @ts-ignore
+        baz: undefined,
+        env: {
+          baz: undefined,
+        },
+      }, (done) => {
+        cy.switchToDomain('foobar.com', [fnName], ([fnName]) => {
+          setTimeout(() => {
+            // this value STILL gets set, but will be blown away on the next switchToDomain with whatever exists in the primary
+            Cypress[fnName]('baz', 'qux')
+          }, 100)
+
+          Cypress[fnName]('baz', 'quux')
+        })
+
+        const isUndefined = Cypress[fnName]('baz')
+
+        expect(isUndefined).to.be.undefined
+
+        cy.wrap({}).then(() => {
+          const isQuux = Cypress[fnName]('baz')
+
+          expect(isQuux).to.equal('quux')
+          done()
+        })
+      })
+
+      it('overwrites different values in secondary if one exists in the primary', {
+        // @ts-ignore
+        baz: 'quux',
+        env: {
+          baz: 'quux',
+        },
+      }, () => {
+        cy.switchToDomain('foobar.com', [fnName], ([fnName]) => {
+          // in previous test, 'baz' was set to 'qux' after the callback window was closed.
+          // this should be overwritten by 'quux' that exists in the primary
+          const isNowQuux = Cypress[fnName]('baz')
+
+          expect(isNowQuux).to.equal('quux')
+        })
+      })
+
+      it(`overwrites different values in secondary, even if the Cypress.${fnName}() value does not exist in the primary`, {
+        // @ts-ignore
+        baz: undefined,
+        env: {
+          baz: undefined,
+        },
+      }, () => {
+        cy.switchToDomain('foobar.com', [fnName], ([fnName]) => {
+          const isNowUndefined = Cypress[fnName]('baz')
+
+          expect(isNowUndefined).to.be.undefined
+        })
+      })
+    })
+
+    context('unserializable', () => {
+      const unserializableFunc = () => undefined
+
+      it('does not sync unserializable values from the primary to the secondary', () => {
+        Cypress[fnName]('unserializable', unserializableFunc)
+        cy.switchToDomain('foobar.com', [fnName], ([fnName]) => {
+          const isUndefined = Cypress[fnName]('unserializable')
+
+          expect(isUndefined).to.be.undefined
+        })
+      })
+
+      it('does not overwrite unserializable values in the primary when none exist in the secondary', () => {
+        Cypress[fnName]('unserializable', unserializableFunc)
+        cy.switchToDomain('foobar.com', () => undefined)
+
+        const isFunc = Cypress[fnName]('unserializable')
+
+        expect(isFunc).to.equal(unserializableFunc)
+      })
+
+      it('overwrites unserializable values in the primary when serializable values of same key exist in secondary', () => {
+        Cypress[fnName]('unserializable', unserializableFunc)
+        cy.switchToDomain('foobar.com', [fnName], ([fnName]) => {
+          Cypress[fnName]('unserializable', undefined)
+        })
+
+        cy.wrap({}).then(() => {
+          const isNowUndefined = Cypress[fnName]('undefined')
```

Comment:
```suggestion
          const isNowUndefined = Cypress[fnName]('unserializable')
```

Reply:
whoops 😅 . Fixed in b7258276e6134ccda3642b3936e8330370b26f03